tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -214,12 +214,14 @@ def raise_ie_not_found():
         log.error(e)
         raise_ie_not_found()
 
-    if ('data_type' in argv and argv.data_type in ['FP16', 'half']) or \
-            ('compress_to_fp16' in argv and argv.compress_to_fp16 is True):
-        argv.data_type = 'FP32'
-        argv.compress_fp16 = True
-    else:
+    # Turn off compression only if it's disabled explicitly by --compress_to_fp16=False or --data_type=FP32.
+    # By default, in all other cases compression is enabled
+    if ('data_type' in argv and argv.data_type in ['FP32', 'float']) or \
+            ('compress_to_fp16' in argv and argv.compress_to_fp16 is False):
         argv.compress_fp16 = False
+    else:
+        argv.compress_fp16 = True
+    argv.data_type = 'FP32'  # if compression was enabled will be restored back to 'FP16' after apply_offline_transformations
```

Comment:
let us remove `argv.data_type = 'FP32'`. Why still we have it even after adjusting e2e tests?

Reply:
e2e test haven't been adjusted to the new argument `compress_to_fp16` yet and still use `data_type`, e.g. one the latest precommits
https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/ie/job/e2e-tests-linux-ubuntu20-e2e/38664/e2e-tests/

let's don't touch this argv.data_type and remove it right after e2e will move to the newest `compress_to_fp16`?